backend-worker/worker.py

```diff
@@ -1,26 +1,546 @@
+        yoff = offset_dict['yoff']
+
+        wallpaper_x = xoff % wallpaper_array.shape[1]
+        wallpaper_y = yoff % wallpaper_array.shape[0]
+
+        win_ysize = offset_dict['win_ysize']
+        win_xsize = offset_dict['win_xsize']
+        wallpaper_x_repeats = (
+            1 + ((wallpaper_x+win_xsize) // wallpaper_array.shape[1]))
+        wallpaper_y_repeats = (
+            1 + ((wallpaper_y+win_ysize) // wallpaper_array.shape[0]))
+        wallpaper_tiled = numpy.tile(
+            wallpaper_array,
+            (wallpaper_y_repeats, wallpaper_x_repeats))[
+            wallpaper_y:wallpaper_y+win_ysize,
+            wallpaper_x:wallpaper_x+win_xsize]
+
+        target_array = numpy.where(
+            parcel_mask_array == 1,
+            wallpaper_tiled,
+            base_array)
+
+        target_band.WriteArray(target_array, xoff=xoff, yoff=yoff)
+
+    target_raster.BuildOverviews()  # default settings for overviews
+
+    shutil.rmtree(working_dir)
+
+
+def pixelpercents_under_parcel(parcel_wkt_epsg3857, source_raster_path):
+    """Get a breakdown of the percent of pixels under a parcel per lulc code.
+
+    Args:
+        parcel_wkt_epsg3857 (str): The parcel WKT in web mercator.
+        source_raster_path (str): The LULC to get pixel counts from.
+
+    Returns:
+        counts (dict): A dict mapping int lulc codes to float (0-1) percent of
+            pixels rounded to 4 decimal places.  This percentage reflects the
+            percentage of pixels under the parcel, not the lulc, so if a parcel
+            covers 4 pixels, 1 of lulc code 5 and 3 of lulc code 6, ``counts``
+            would be ``{5: 0.25, 6: 0.75}``.
+    """
+    if source_raster_path.startswith(('https', 'http')):
+        source_raster_path = f'/vsicurl/{source_raster_path}'
+    source_raster = gdal.OpenEx(source_raster_path,
+                                gdal.GA_ReadOnly | gdal.OF_RASTER)
+    source_band = source_raster.GetRasterBand(1)
+    geotransform = source_raster.GetGeoTransform()
+    inv_geotransform = gdal.InvGeoTransform(geotransform)
+
+    parcel = _reproject_to_nlud(parcel_wkt_epsg3857)
+    # Convert lon/lat degrees to x/y pixel for the dataset
+    minx, miny, maxx, maxy = parcel.bounds
+    _x0, _y0 = gdal.ApplyGeoTransform(inv_geotransform, minx, miny)
+    _x1, _y1 = gdal.ApplyGeoTransform(inv_geotransform, maxx, maxy)
+    x0, y0 = min(_x0, _x1), min(_y0, _y1)
+    x1, y1 = max(_x0, _x1), max(_y0, _y1)
+
+    pygeoprocessing.geoprocessing.shapely_geometry_to_vector(
+        [parcel],
+        os.path.join('parcel_loaded.fgb'),
+        _WEB_MERCATOR_SRS.ExportToWkt(), 'FlatGeoBuf')
+
+    # "Round up" to the next pixel
+    x0 = math.floor(x0)
+    y0 = math.floor(y0)
+    x1 = math.ceil(x1)
+    y1 = math.ceil(y1)
+    array = source_band.ReadAsArray(
+        int(x0), int(y0), int(x1-x0), int(y1-y0))
+
+    # create a new in-memory dataset filled with 0
+    gdal_driver = gdal.GetDriverByName('MEM')
```

Comment:
Been awhile since I've seen 'MEM' raster datasets in action!

Reply:
They are really handy for certain things!  I think this is a good use of them, although I suppose we risk possibly exhausting memory if a bbox is extra large.